src/fight/mod.rs

```diff
@@ -3,157 +3,280 @@
+    case: &ComputeCase,
+) -> (usize, f64) {
+    let attacking_stats: &model::Stats = attacking_regiment.get_model().get_stats();
+    let nb_touch: usize =
+        (attacking_stats.get_attack() as f64 * 1.5 * attacking_regiment.get_cols() as f64).round()
+            as usize;
+    let wound_probability: f64 = computation_tools::compute_damage_probability(
+        attacking_stats,
+        defending_regiment.get_model().get_stats(),
     );
-    let nb_attacks: usize = get_nb_attacks(regiment_attacking);
-    let nb_wounds: usize = get_nb_wounds(nb_attacks, to_hit, to_wound);
-    let final_result: usize = get_final_result(
-        nb_wounds,
-        regiment_defending_stats.get_armour(),
-        regiment_attacking_stats.get_armour_penetration(),
+    let max_hit: usize = std::cmp::min(nb_touch, defending_regiment.get_regiment_health_points());
+    let checkpoints: (usize, usize) = find_great_gauss_checkpoints(
+        nb_touch,
+        wound_probability,
+        defending_regiment.get_regiment_health_points(),
     );
-    final_result
+
+    if let ComputeCase::MEAN = case {
+        return compute_mean_case(attacking_regiment, defending_regiment);
+    }
+    match case {
+        ComputeCase::BEST => (
+            checkpoints.1,
+            evaluate_gauss_interval(checkpoints.1, max_hit, nb_touch, wound_probability),
+        ),
+        ComputeCase::WORST => (
+            checkpoints.0,
+            evaluate_gauss_interval(0, checkpoints.0, nb_touch, wound_probability),
+        ),
+        ComputeCase::MEAN => unreachable!("Code not supposed to be reached!"),
+    }
 }
 
-/// Return the fastest Model between 2 Model
-/// If the first Model is faster than the second Model, return 1
-/// If the second Model is faster than the first Model, return 2
-/// If the Models have the same speed, return 0
+/// Make two units fight and return the probability that the fight occurs during the game
 ///
-/// # Parameters
-/// attacking_model (&model::Model): The first Model
+/// ## Parameters
+/// (&mut regiment::Regiment, ComputeCase) fastest: The fastest regiment
 ///
-/// defending_model (&model::Model): The second Model
+/// (&mut regiment::Regiment, ComputeCase) slowest: The slowest regiment
 ///
-/// # Return
-/// usize: The fastest Model
-fn find_the_fastest(attacking_model: &model::Model, defending_model: &model::Model) -> usize {
-    let attacking_model_agility: usize = attacking_model.get_stats().get_agility() + 1;
-    let defending_model_agility: usize = defending_model.get_stats().get_agility();
-
-    match attacking_model_agility.cmp(&defending_model_agility) {
-        std::cmp::Ordering::Greater => 1,
-        std::cmp::Ordering::Less => 2,
-        std::cmp::Ordering::Equal => 0,
+/// (bool) speed_equality: A boolean value to specify if the regiment have the same speed or not
+///
+/// ## Return
+/// Prediction: The prediction computed according to the specified Compute Case
+fn rumble(
```

Comment:
<img width="498" alt="image" src="https://user-images.githubusercontent.com/57897025/195459406-10dc8e3e-8e3a-4872-b112-079cb34c4ccc.png">
Why this name ?